botocore/loaders.py

```diff
@@ -355,7 +372,32 @@ def load_service_model(self, service_name, type_name, api_version=None):
             api_version = self.determine_latest_version(
                 service_name, type_name)
         full_path = os.path.join(service_name, api_version, type_name)
-        return self.load_data(full_path)
+        model = self.load_data(full_path)
+
+        if not load_extras:
+            return model
+
+        # Load in all the extras
+        for tool_name in self._tool_extras:
+            self._load_extras(
+                model, service_name, type_name, api_version, tool_name)
+
+        return model
+
+    def _load_extras(self, service_model, service_name, type_name,
+                     api_version, tool_name):
+        """Load and process an extras file into a model."""
+        extras_type_name = '%s.%s-extras' % (type_name, tool_name)
+        full_path = os.path.join(service_name, api_version, extras_type_name)
+
+        try:
+            extra_data = self.load_data(full_path)
+        except DataNotFoundError:
+            return
+
+        merge_data = extra_data.get('merge')
+        if merge_data is not None:
+            deep_merge(service_model, merge_data)
```

Comment:
I don't think you're testing the case where the `merge` key does not exist.

FWIW, I would prefer that the "processing" logic is delegated to another class.  Even in the simple case of a single merge key, it's somewhat difficult to just isolate the "merge" logic, and I can only see this logic growing in the future (e.g `add`, `remove`, etc).

For now though we should at least get a test added for `merge` not existing (or make it required)